app/models/user.rb

```diff
@@ -226,15 +201,14 @@ def perishable_token_expired?
     perishable_token && updated_at < (Time.now - User.perishable_token_valid_for)
   end
 
-  def require_email_verification
-    set_approval_status_to_pending_email_verification
-    self.requires_manual_approval = !has_government_affiliated_email?
-    true
-  end
-
   def self.from_omniauth(auth)
     find_or_create_by(email: auth.info.email).tap do |user|
       user.update(uid: auth.uid)
+      if user.requires_manual_approval?
+        user.set_approval_status_to_pending_approval
+      else
+        user.set_approval_status_to_approved
+      end
```

Comment:
This logic should not run in the `from_omniauth` method, because we don't need to run this logic every time a user logs in. The logic should probably be in the `set_initial_approval_status` method, or one of the related methods that only runs during account creation.

Reply:
Let's chat. I still want to leave this till I dig into that section.